src/app/component/CartCoupon/CartCoupon.component.js

```diff
@@ -0,0 +1,117 @@
+import { PureComponent } from 'react';
+import Field from 'Component/Field';
+import Loader from 'Component/Loader';
+import PropTypes from 'prop-types';
+import './CartCoupon.style';
+
+class CartCoupon extends PureComponent {
+    static propTypes = {
+        isLoading: PropTypes.bool.isRequired,
+        couponCode: PropTypes.string,
+        handleApplyCouponToCart: PropTypes.func.isRequired,
+        handleRemoveCouponFromCart: PropTypes.func.isRequired
+    };
+
+    static defaultProps = {
+        couponCode: ''
+    };
+
+    state = {
+        enteredCouponCode: ''
+    };
+
+    handleCouponCodeChange = this.handleCouponCodeChange.bind(this);
+
+    handleApplyCoupon = this.handleApplyCoupon.bind(this);
+
+    handleRemoveCoupon = this.handleRemoveCoupon.bind(this);
+
+    handleCouponCodeChange(enteredCouponCode) {
+        this.setState({
+            enteredCouponCode
+        });
+    }
+
+    handleApplyCoupon() {
+        const { handleApplyCouponToCart } = this.props;
+        const { enteredCouponCode } = this.state;
+
+        handleApplyCouponToCart(enteredCouponCode);
+    }
+
+    handleRemoveCoupon() {
+        const { handleRemoveCouponFromCart } = this.props;
+
+        handleRemoveCouponFromCart();
+
+        // We need to reset input field. If we do it in applyCouponCode,
+        // then it will disappear if code is incorrect. We want to avoid it
+        this.setState({
+            enteredCouponCode: ''
+        });
+    }
+
+    renderApplyCoupon() {
+        const { enteredCouponCode } = this.state;
+
+        return (
+            <>
+                <Field
+                  type="text"
+                  id="couponCode"
+                  name="couponCode"
+                  value={ enteredCouponCode }
+                  placeholder={ __('Coupon Code') }
+                  onChange={ this.handleCouponCodeChange }
+                  mix={ { block: 'CartCoupon', elem: 'Input' } }
+                />
+                <button
+                  block="CartCoupon"
+                  elem="Button"
+                  mix={ { block: 'Button' } }
+                  disabled={ !enteredCouponCode }
+                  onClick={ this.handleApplyCoupon }
+                >
+                    { __('Apply Coupon') }
+                </button>
+            </>
+        );
+    }
+
+    renderRemoveCoupon() {
+        const { couponCode } = this.props;
+
+        return (
+            <>
+                <p block="CartCoupon" elem="Message">
+                    { __('Applied coupon code') }
```

Comment:
`__('Applied coupon code: ')` you can put `:` into translation.

Reply:
Resolved